timetracker/timetracker/urls.py

```diff
@@ -16,12 +16,22 @@
 from django.conf.urls import include, url
 from django.contrib import admin
 from django.http import HttpResponse
+import random
 
+def hello (request):
 
-def hello_world(request):
-    return HttpResponse("Hello world")
+    name=request.GET.get("name","default")
+    return HttpResponse('<b>hello</b> '+name)
 
+def displayform (request):
+    return HttpResponse("""<form action="hello">
+First name:<br>
+<input type="text" name="name" value="Noob">
+<br>
+<input type="submit" value="Submit">
+</form> """)
 urlpatterns = [
-    url(r'^$', hello_world),
+    url(r'^$', displayform),
+    url(r'^hello$', hello),
```

Comment:
We'll make mention of it in class this afternoon, but there's no need to have two views here - the form can submit to the same URL it's being served from (and we could put an `if` in there to check for the GET value).